src/OlStyleParser.ts

```diff
@@ -910,15 +910,16 @@ export class OlStyleParser implements StyleParser {
    * @return {object} The OL Style object
    */
   getOlPolygonSymbolizerFromFillSymbolizer(symbolizer: FillSymbolizer) {
+    const fill = symbolizer.color ? new this.OlStyleFillConstructor({
+      color: (symbolizer.color && symbolizer.opacity !== null && symbolizer.opacity !== undefined) ?
```

Comment:
If  `OlStyleUtil.getRgbaColor` had a default value of  `1` for the opacity things would be easier here. I'd prefer to see a change in the util method.

Reply:
Actually, I'm trying to handle the case of having no fill symbolizer here, in case the color is not set. So moving this into the util will not work since the fill symbolizer would be created anyway.

Reply:
But your checking for the opacity here `&& symbolizer.opacity !== null && symbolizer.opacity`. :thinking: This could be removed if `getRgbaColor` would be smarter. So it could be become:
```typescript
const fill = symbolizer.color ? new this.OlStyleFillConstructor({
      color: OlStyleUtil.getRgbaColor(symbolizer.color, symbolizer.opacity)
}) : null;
```

Reply:
Oh yes, that's just copied from the original code. I've factored out the opacity check into the util now.

Reply:
Oh yes, that's just copied from the original code. I've factored out the opacity check into the util now.

Reply:
Sorry, I'm reverting that one since it breaks almost all tests.

Reply:
Hmpf. Ok.

Would be cool if you could open an issue for this.

Reply:
Done in https://github.com/terrestris/geostyler/issues/894